### PR TITLE
Update java invoker spring version to 3.0.0

### PR DESCRIPTION
**Pull request by joeeltgroth**

### CODE REVIEW

**Review by dolfolife — APPROVED** [review]
LGTM

File: DEPLOYING.md [hunk 1]
```diff
@@ -15,6 +15,11 @@ Examples:
   * Python: BP_FUNCTION=func.main. `func` is the name of the .py file. main is the `method`.
   * Java: BP_FUNCTION=function.Handler. `function` the package. `Handler` is the class that implements Function.
 
+Note:
```
Comment (menehune23) [inline DEPLOYING.md:18]: This is a nit, but "notes" generally look nice in a quote block

**Review by menehune23 — CHANGES_REQUESTED** [review]
Just one nitpick (feel free to ignore) and another concern about the open vs non-open JDK. If you have a definitive answer on the JDK, feel free to re-request review.

File: DEPLOYING.md (hunk 1 above)
Comment (joeeltgroth) [reply DEPLOYING.md]: Yes, good idea. Thanks!

File: invokers/java/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Copyright 2021-2022 VMware, Inc.
 # SPDX-License-Identifier: BSD-2-Clause
 
-FROM maven:3.8.3-jdk-11
+FROM maven:3.8.7-openjdk-18
```
[the remark this answers is not in the document]
Comment (joeeltgroth) [reply invokers/java/Dockerfile:4]: I don't believe there will be an issue with OpenJDK.  The maven image used in our Dockerfile switches from jdk to openjdk after version 14.   OpenJDK is the reference implementation and GPL-licensed.

**Pull request merged**